src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -149,14 +151,17 @@ const std::shared_ptr<AbstractLQPNode>& SQLPipelineStatement::get_optimized_logi
 
   auto optimizer_rule_durations = std::make_shared<std::vector<OptimizerRuleMetrics>>();
 
-  _optimized_logical_plan = _optimizer->optimize(std::move(unoptimized_lqp), optimizer_rule_durations);
+  auto [result_lqp, result_context] =
+      _optimizer->optimize_with_context(std::move(unoptimized_lqp), optimizer_rule_durations);
+  _optimized_logical_plan = result_lqp;
+  _optimization_context = std::move(result_context);
```

Comment:
Can we just do the following?
```suggestion
  [_optimized_logical_plan, _optimization_context] =
      _optimizer->optimize_with_context(std::move(unoptimized_lqp), optimizer_rule_durations);
```